stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+
+        Parameters
+        ----------
+        prior_model: jaxns.prior.PriorModelType object
+            A prior generator object
+
+        likelihood_model: jaxns.types.LikelihoodType object
+            A likelihood fucntion which takes in the arguments of the prior
+            model and returns the loglikelihood of the model
+
+        max_samples: int, default 1e4
+            The maximum number of samples to be taken by the nested sampler
+
+        Returns
+        ----------
+        Results: jaxns.results.NestedSamplerResults object
+            The results of the nested sampling process
+
+        """
+        if not jax_avail:
+            raise ImportError("Jax is required")
+
+        if not can_sample:
+            raise ImportError("Jaxns not installed! Can't sample!")
+
+        self.prior_model = prior_model
+        self.likelihood_model = likelihood_model
+
+        NSmodel = Model(prior_model=self.prior_model, log_likelihood=self.likelihood_model)
+        NSmodel.sanity_check(random.PRNGKey(10), S=100)
+
+        self.Exact_ns = ExactNestedSampler(NSmodel, num_live_points=500, max_samples=max_samples)
+        Termination_reason, State = self.Exact_ns(
+            random.PRNGKey(42), term_cond=TerminationCondition(live_evidence_frac=1e-4)
+        )
+        self.Results = self.Exact_ns.to_results(State, Termination_reason)
+        print("Simulation Complete")
+
+    def get_evidence(self):
+        """
+        Returns the log evidence of the model
+        """
+        return self.Results.log_Z_mean
+
+    def print_summary(self):
+        """
+        Prints a summary table for the model parameters
+        """
+        self.Exact_ns.summary(self.Results)
+
+    def plot_diagnostics(self):
+        """
+        Plots the diagnostic plots for the sampling process
+        """
+        self.Exact_ns.plot_diagnostics(self.Results)
+
+    def plot_cornerplot(self):
+        """
+        Plots the corner plot for the sampled hyperparameters
+        """
+        self.Exact_ns.plot_cornerplot(self.Results)
+
+    def get_parameters_names(self):
+        """
+        Returns the names of the parameters
+        """
+        return sorted(self.Results.samples.keys())
+
+    def get_max_posterior_parameters(self):
+        """
+        Returns the optimal parameters for the model based on the NUTS sampling
+        """
+        max_post_idx = jnp.argmax(self.Results.log_posterior_density)
+        map_points = jax.tree_map(lambda x: x[max_post_idx], self.Results.samples)
+
+        return map_points
+
+    def get_max_likelihood_parameters(self):
+        """
+        Retruns the maximum likelihood parameters
+        """
+        max_like_idx = jnp.argmax(self.Results.log_L_samples)
+        max_like_points = jax.tree_map(lambda x: x[max_like_idx], self.Results.samples)
+
+        return max_like_points
+
+    def posterior_plot(self, name: str, n=0, axis=None, save=False, filename=None):
```

Comment:
I think it would be great if this plotting could be moved to an object-oriented version of matplotlib, and include an `ax` parameter that allows the user to plot this into an existing `matplotlib.Axes` object, so that one can easily make multi-panel plots using this method